pkg/bootkube/approve.go

```diff
@@ -0,0 +1,88 @@
+package bootkube
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api/v1"
+	certificates "k8s.io/kubernetes/pkg/apis/certificates/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5/typed/certificates/v1alpha1"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/util/wait"
+)
+
+// ApproveKubeletCSRs indefinitely approves any CSRs submitted by Kubelet
+// instances, that are in the process of bootstrapping their TLS assets, without
+// making any kind of validation.
+func ApproveKubeletCSRs() error {
+	errCh := make(chan error)
+
+	client, err := clientset.NewForConfig(
+		&restclient.Config{Host: insecureAPIAddr},
+	)
+	if err != nil {
+		return err
+	}
+
+	watchList := cache.NewListWatchFromClient(
+		client.CertificatesV1alpha1().RESTClient(),
+		"certificatesigningrequests",
+		v1.NamespaceAll,
+		fields.Everything(),
+	)
+
+	_, controller := cache.NewInformer(
+		watchList,
+		&certificates.CertificateSigningRequest{},
+		time.Second*5,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
```

Comment:
Sounds unlikely but worth doing! Good catch. I added a new commit and verified that:
1) It validates CSRs that were submitted before the watcher started.
2) It doesn't validate CSRs that have already been approved or denied.